chaininfo/usecase/chain_info_usecase.go

```diff
@@ -97,20 +104,46 @@ func (p *chainInfoUseCase) OnPricingUpdate(ctx context.Context, height uint64, b
 	return nil
 }
 
+// OnPoolLiquidityCompute implements domain.PoolLiquidityComputeListener.
+func (p *chainInfoUseCase) OnPoolLiquidityCompute(height int64) error {
+	p.poolLiquidityUpdateHeightMx.Lock()
+	defer p.poolLiquidityUpdateHeightMx.Unlock()
+	p.latestPoolLiquidityUpdateHeight = uint64(height)
+
+	return nil
+}
+
 // ValidatePriceUpdates implements mvc.ChainInfoUsecase.
 func (p *chainInfoUseCase) ValidatePriceUpdates() error {
 	p.priceUpdateHeightMx.RLock()
 	latestPriceUpdateHeight := p.latestPricesUpdateHeight
 	p.priceUpdateHeightMx.RUnlock()
 
-	// Check that the initial prices have been computed and received.
-	if latestPriceUpdateHeight == initialPriceUpdateHeight {
-		return errors.New("healthcheck has not received initial price updates")
+	return validateUpdate(latestPriceUpdateHeight, p.lastIngestedHeight, pricingUpdateName)
+}
+
+// ValidatePriceUpdates implements mvc.ChainInfoUsecase.
+func (p *chainInfoUseCase) ValidatePoolLiquidityUpdates() error {
+	p.priceUpdateHeightMx.RLock()
+	latestPoolLiquidityUpdateHeight := p.latestPoolLiquidityUpdateHeight
+	p.priceUpdateHeightMx.RUnlock()
+
+	return validateUpdate(latestPoolLiquidityUpdateHeight, p.lastIngestedHeight, poolLiquidityPricingUpdateName)
+}
+
+// validateUpdate validates the update for the given update name, current update height, and latest ingested height.
+// It returns an error if the update is invalid.
+// The update is invalid if the current update height is less than the latest ingested height minus the update height buffer.
+// The update is also invalid if the current update height is equal to the initial update height.
+func validateUpdate(currentUpdateHeight uint64, latestIngestedHeight uint64, updateName string) error {
+	// Check that the initial pool liquidities have been computed and received.
+	if currentUpdateHeight == initialUpdateHeight {
+		return fmt.Errorf("healthcheck has not received initial %s updates", updateName)
 	}
 
-	// Check that the price updates have been occurring
-	if latestPriceUpdateHeight < p.lastSeenUpdatedHeight-priceUpdateHeightBuffer {
-		return errors.New("latest price update height is less than the last seen updated height")
+	// Check that the pool liquidity updates have been occurring
+	if currentUpdateHeight < latestIngestedHeight-updateHeightBuffer {
```

Comment:
Space before and after the minus ```-``` sign for readability?

Reply:
This is a standard go formatting. It would auto-set it back to the current way if I were to change this.

Let's generally aim to set up linters for any style-related rules